app/cafe.py

```diff
@@ -0,0 +1,19 @@
+from app.errors import NotVaccinatedError, \
+                        OutdatedVaccineError, \
+                        NotWearingMaskError
+import datetime
+
+
+class Cafe:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def visit_cafe(self, vstr: dict) -> str:
+        if "vaccine" not in vstr.keys() or vstr["vaccine"] == None:
```

Comment:
Dict has get method, the second argument is value to be returned in case of missing key, by default it's None, so you can use it to write such nice shortly check for that:
```
expiration_date = vstr.get("vaccine", {}).get("expiration_date")
if not expiration_date: 
    raise NotVaccinatedError("Good error message")
elif expiration_date < datetime.date.today():
    raise OutdatedVaccineError("Good error message")
...
```